client/src/streams.rs

```diff
@@ -23,10 +23,11 @@ impl Streams {
         self.streams.clear();
     }
 
-    pub fn add_packet(&mut self, packet: Packet) {
+    pub fn add_packet(&mut self, mut packet: Packet) {
         let is_new = self.packets.is_new(&packet);
 
         if is_new {
+            self.detect_previous_packet_lost(&mut packet);
```

Comment:
Did you consider that packets can arrive not in order? Let's say that packet `x+1` arrived, and only then packet `x` arrived, will the `previous_packet_is_lost` flag be updated for packet `x+1`?

Reply:
Why use weird heuristic? Just, for every incoming packet, if the next packet is present, set it's `previous_packet_lost` to false.

Reply:
This still can miss, so let's leave a `FIXME` comment there, but for now I guess it's alright.